test/functional/wallet_basic.py

```diff
@@ -350,6 +403,87 @@ def run_test(self):
+                self.nodes[2].sendtoaddress,
+                address=address,
+                amount=1.0,
+                conf_target=-1,
+                estimate_mode='btc/kb')
+            txid = self.nodes[2].sendtoaddress(
+                address=address,
+                amount=1.0,
+                conf_target=0.00002500,
+                estimate_mode='btc/kb',
+            )
+            tx_size = self.get_vsize(self.nodes[2].gettransaction(txid)['hex'])
+            self.sync_all(self.nodes[0:3])
+            self.nodes[0].generate(1)
+            self.sync_all(self.nodes[0:3])
+            postbalance = self.nodes[2].getbalance()
+            fee = prebalance - postbalance - Decimal('1')
+            assert_fee_amount(fee, tx_size, Decimal('0.00002500'))
+
+            # send with explicit sat/b fee
+            self.sync_all(self.nodes[0:3])
+            self.log.info("test explicit fee (sendtoaddress as sat/b)")
+            self.nodes[0].generate(1)
+            prebalance = self.nodes[2].getbalance()
+            assert prebalance > 2
+            address = self.nodes[1].getnewaddress()
+            # Throw if no conf_target provided
+            assert_raises_rpc_error(-8, "Selected estimate_mode requires a fee rate",
+                self.nodes[2].sendtoaddress,
+                address=address,
+                amount=1.0,
+                estimate_mode='SAT/b')
+            # Throw if negative feerate
+            assert_raises_rpc_error(-3, "Amount out of range",
+                self.nodes[2].sendtoaddress,
+                address=address,
+                amount=1.0,
+                conf_target=-1,
+                estimate_mode='SAT/b')
+            assert_raises_rpc_error(-4, "Fee rate is too low for the current mempool. Increase it to 1.000 sat/B or wait.",
+                self.nodes[2].sendtoaddress,
+                address=address,
+                amount=1.0,
+                conf_target=0.2,
+                estimate_mode='SAT/B')
+            txid = self.nodes[2].sendtoaddress(
+                address=address,
+                amount=1.0,
+                conf_target=2,
+                estimate_mode='SAT/B',
+            )
+            tx_size = self.get_vsize(self.nodes[2].gettransaction(txid)['hex'])
+            self.sync_all(self.nodes[0:3])
+            self.nodes[0].generate(1)
+            self.sync_all(self.nodes[0:3])
+            postbalance = self.nodes[2].getbalance()
+            fee = prebalance - postbalance - Decimal('1')
+            assert_fee_amount(fee, tx_size, Decimal('0.00002000'))
+
+            # Import address and private key to check correct behavior of spendable unspents
+            # 1. Send some coins to generate new UTXO
+            address_to_import = self.nodes[2].getnewaddress()
+            txid = self.nodes[0].sendtoaddress(address_to_import, 1)
+            self.nodes[0].generate(1)
+            self.sync_all(self.nodes[0:3])
+
```

Comment:
I believe this was a mis-rebase. This test goes back to 2016, and is above the added tests (prior to the rebase, it was here, below the added tests)

Reply:
Yes, that is correct! Good catch. Fixed.